darts/ad/score.py

```diff
@@ -0,0 +1,567 @@
+        super()._check_norm()
+        self.model = GaussianMixture(n_components=n_components)
+
+    def _fit_core(self, series: Union[TimeSeries, Sequence[TimeSeries]]):
+
+        self._fit_called = True
+        self.model.fit(series.all_values().flatten().reshape(-1, 1))
+
+    def _compute_core(
+        self, series: Union[TimeSeries, Sequence[TimeSeries]]
+    ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+
+        np_anomaly_score = np.exp(
+            self.model.score_samples(series.all_values().flatten().reshape(-1, 1))
+        )
+        return TimeSeries.from_times_and_values(series._time_index, np_anomaly_score)
+
+
+class KmeansScorer(TrainableScorer):
+    """Kmean anomaly score"""
+
+    def __init__(
+        self,
+        characteristic_length: Optional[int] = None,
+        k: Union[int, list[int]] = 2,
+        distance_function="l1",
+    ) -> None:
+        super().__init__(characteristic_length=characteristic_length)
+        self.k = k
+        self.distance_function = distance_function
+        self._check_norm()
+        self.model = KMeans(n_clusters=k)
+
+    def _fit_core(self, series: Union[TimeSeries, Sequence[TimeSeries]]):
+
+        self._fit_called = True
+        self.model.fit(series.all_values().flatten().reshape(-1, 1))
+
+    def _compute_core(
+        self, series: Union[TimeSeries, Sequence[TimeSeries]]
+    ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+        # return distance to the clostest centroid
+        np_anomaly_score = self.model.transform(
+            series.all_values().flatten().reshape(-1, 1)
+        ).min(axis=1)
+        return TimeSeries.from_times_and_values(series._time_index, np_anomaly_score)
+
+
+class WasserteinScorer(TrainableScorer):
+    """WasserteinScorer anomaly score"""
+
+    def __init__(
+        self, characteristic_length: Optional[int] = None, distance_function="l1"
+    ) -> None:
+        if characteristic_length is None:
+            characteristic_length = 10
+        super().__init__(characteristic_length=characteristic_length)
+        self.distance_function = distance_function
+        super()._check_norm()
+
+        raise_if(
+            self.characteristic_length == 0,
+            "characteristic_length must be stricly higher than 0,"
+            "(preferably higher than 10 as it is the number of samples of the test distribution)",
+        )
+
+    def _fit_core(self, series: Union[TimeSeries, Sequence[TimeSeries]]):
+
+        self._fit_called = True
+        self.training_data = series.all_values().flatten()
+
+    def _compute_core(
+        self, series: Union[TimeSeries, Sequence[TimeSeries]]
+    ) -> Union[TimeSeries, Sequence[TimeSeries]]:
+        # return distance to the clostest centroid
+        distance = []
+
+        for i in range(len(series) - self.characteristic_length + 1):
+            distance.append(
+                wasserstein_distance(
+                    self.training_data,
```

Comment:
Does it make sense to have a window for the training data? 

- The idea here is to compare the distribution of the residuals of the training data vs a window w in the testing data. (here we would be talking about residuals). For now, we take the entire training data. If we only take a window of the training data, we reduce the number of samples (however, the more, the better because we would like to have the true distribution of our training data). 
- Secondly, if we go with this idea of using a window for the training data, what would be the rule to compare the train and test distribution? Should we compare the window in the training and testing that have identical timestamps? But if the series is stationary (does not depend on time) then it doesn't make sense to compare at the same time range. The signal in the training data at time 5 to 15 has maybe nothing to do with the signal in the testing data at time 5 to 15. 
- Thirdly, what about if we train with a list of time series? 

Were you thinking of partitioning the training data into n window of size w and averaging the distributions? So that in the Wasserstein distance, we compare the same number of samples?